pkg/stanza/fileconsumer/reader_test.go

```diff
@@ -239,3 +240,121 @@ func TestMapCopy(t *testing.T) {
 	assert.Equal(t, 1, initMap["intVal"])
 	assert.Equal(t, "OrigStr", initMap["strVal"])
 }
+
+func TestReaderUpdateFingerprint(t *testing.T) {
+	testCases := []struct {
+		name            string
+		fingerprintSize int
+		bufferSize      int
+		initBytes       []byte
+		moreBytes       []byte
+		endOffset       int64
+	}{
+		{
+			name:            "new_file",
+			fingerprintSize: 10,
+			bufferSize:      100,
```

Comment:
How is `bufferSize` used in tests?
Maybe I missed something but I see that now size of the buffer is always set to [defaultBufSize](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8dc41a8582dae9680e49c8fa997fac4fc1487be7/pkg/stanza/fileconsumer/scanner.go#LL29C25-L29C39),  see [this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8dc41a8582dae9680e49c8fa997fac4fc1487be7/pkg/stanza/fileconsumer/scanner.go#LL29C25-L29C39), and  [testReaderFactory](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8dc41a8582dae9680e49c8fa997fac4fc1487be7/pkg/stanza/fileconsumer/reader_test.go#L196-L210) don't set the `bufferSize` 🤔

Reply:
You are right. I was originally testing `Reader.Read` directly and was using this.. Taking another look to incorporate this somehow as I think it's important given the bug.